File: patch_supabase.js
const fs = require('fs');

// Patch lib/supabase.ts
const supabasePath = 'lib/supabase.ts';
let supabaseContent = fs.readFileSync(supabasePath, 'utf8');

const throwBlock = `// Production: strict enforcement. Preview/dev: optional with warning
if (!isConfigured && isProduction) {
  throw new Error(
    'Missing Supabase configuration in production. NEXT_PUBLIC_SUPABASE_URL and NEXT_PUBLIC_SUPABASE_ANON_KEY must be set.'
  );
}`;

supabaseContent = supabaseContent.replace(throwBlock, `// Soft warning to prevent crashing the Vercel static analyzer during build.
if (!isConfigured && isProduction) {
  console.warn('Missing Supabase configuration in production build step. Next.js will ignore this until runtime.');
}`);
fs.writeFileSync(supabasePath, supabaseContent);

// Patch lib/auth-server.ts
const authServerPath = 'lib/auth-server.ts';
let authServerContent = fs.readFileSync(authServerPath, 'utf8');

authServerContent = authServerContent.replace(
  /const supabaseUrl = process\.env\.NEXT_PUBLIC_SUPABASE_URL!;\nconst supabaseServiceKey = process\.env\.SUPABASE_SERVICE_ROLE_KEY!;/g,
  `const getSupabaseUrl = () => process.env.NEXT_PUBLIC_SUPABASE_URL || '';
const getSupabaseServiceKey = () => process.env.SUPABASE_SERVICE_ROLE_KEY || '';`
);

authServerContent = authServerContent.replace(
  /export const supabaseAdmin = createClient\(supabaseUrl, supabaseServiceKey, \{/g,
  `// Lazy init to prevent build-time crashes
let _adminClient: any = null;
export const getSupabaseAdmin = () => {
  if (!_adminClient) {
    _adminClient = createClient(getSupabaseUrl(), getSupabaseServiceKey(), {`
);

authServerContent = authServerContent.replace(
  /    persistSession: false,\n  \},\n\}\);/g,
  `    persistSession: false,
  }
});
  }
  return _adminClient;
};`
);

authServerContent = authServerContent.replace(
  /return createClient\(supabaseUrl, process\.env\.NEXT_PUBLIC_SUPABASE_ANON_KEY!, \{/g,
  `return createClient(getSupabaseUrl(), process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY || '', {`
);

fs.writeFileSync(authServerPath, authServerContent);

File: patch_supabase.py
import re

with open('lib/supabase.ts', 'r') as f:
    content = f.read()

# Make the module safe to load during build if env vars are missing
content = re.sub(
    r"const isProduction = process\.env\.NEXT_PUBLIC_VERCEL_ENV === 'production';\s*const isConfigured = !!supabaseUrl && !!supabaseAnonKey;\s*// Production: strict enforcement\. Preview/dev: optional with warning\s*if \(!isConfigured && isProduction\) \{\s*throw new Error\(\s*'Missing Supabase configuration in production\. NEXT_PUBLIC_SUPABASE_URL and NEXT_PUBLIC_SUPABASE_ANON_KEY must be set\.'\s*\);\s*\}",
    """const isProduction = process.env.NEXT_PUBLIC_VERCEL_ENV === 'production';

const isConfigured = !!supabaseUrl && !!supabaseAnonKey;

// In Next.js, modules are executed at build time. We don't want to fail the build
// if runtime variables aren't present during the static analysis phase.
// Instead of throwing an error immediately, we log a warning. The application
// will handle the missing client gracefully or fail at runtime when accessed.
if (!isConfigured && isProduction) {
  console.warn('⚠️ Missing Supabase configuration in production build. NEXT_PUBLIC_SUPABASE_URL and NEXT_PUBLIC_SUPABASE_ANON_KEY should be set.');
}""",
    content
)

with open('lib/supabase.ts', 'w') as f:
    f.write(content)
